### PR TITLE
fix(azure): boost the Azure SDK for GO version

### PR DESCRIPTION
**Pull request by mantissahz**
#### Which issue(s) this PR fixes:

Issue # longhorn/longhorn#8965

### CODE REVIEW

File: azblob/azblob.go [hunk 1]
```diff
@@ -194,6 +194,7 @@ func (s *BackupStoreDriver) Download(src, dst string) error {
 	if err != nil {
 		return err
 	}
+	defer rc.Close()
```
Comment (derekbit) [inline azblob/azblob.go:197]: Is it a bug fix?

Comment (mantissahz) [reply azblob/azblob.go:197]: Some kind because we do not close it now.

Comment (derekbit) [reply azblob/azblob.go:197]: Suggest putting it into another PR.

Comment (mantissahz) [reply azblob/azblob.go]: Removed from this PR. Thanks.

Comment (derekbit) [reply azblob/azblob.go]: Can you create a new PR for it? I can review them together. Thank you.

Comment (mantissahz) [reply azblob/azblob.go]: here is the new PR https://github.com/longhorn/backupstore/pull/186

File: azblob/azblob_service.go [hunk 2]
```diff
@@ -130,12 +131,12 @@ func (s *service) putBlob(blob string, reader io.ReadSeeker) error {
 func (s *service) getBlob(blob string) (io.ReadCloser, error) {
```
Comment (derekbit) [inline azblob/azblob_service.go:131]: https://github.com/longhorn/backupstore/blob/master/azblob/azblob.go#L193-L199

`rc` is not closed.

Comment (mantissahz) [reply azblob/azblob_service.go:131]: Yes, separate the commits into two PRs.
Here is the PR https://github.com/longhorn/backupstore/pull/186 to close the `rc` reader.

**Review by derekbit — APPROVED** [review]

**Pull request merged**